### ISSUE TITLE
Incorrect `this trait takes 1 generic argument but 0 generic arguments were supplied`

### ISSUE THREAD

**Issue by Veykril** [rust-lang/rust-analyzer#19669]
In
```rust
    let _: Option<i32> = Some(0).map(Into::into);
```
rust-analyzer complains about the lacking generic argument for `Into` even though that is inferred

**Comment by RaulTrombin** [comment]
Hey @Veykril , found same issue with tracing logs crates: 
all debug, info, trace....macros raise this warning now.
![Image](https://github.com/user-attachments/assets/22a12076-9caa-4bc0-9bce-ac471f632022)

![Image](https://github.com/user-attachments/assets/1551918f-1056-4acb-9fd0-ea3117774f0f)

**Comment by Veykril** [comment]
cc @ChayimFriedman2
